docs/data/toolpad/core/introduction/tutorial.md

```diff
@@ -1,3 +1,105 @@
-# Tutorial
+---
+title: Tutorial
+---
 
-<p class="description">Tutorial</p>
+# Toolpad Core - Tutorial
+
+<p class="description">Learn how to use Toolpad Core through an illustrative example dashboard.</p>
+
+## Bootstrapping
+
+<br/>
+
+1. To choose a project name and create a basic project for this tutorial, run:
+
+<codeblock storageKey="package-manager">
+
+```bash npm
+npx create-toolpad-app@latest --example core-tutorial
+```
+
+```bash pnpm
+pnpm create toolpad-app --example core-tutorial
+```
+
+```bash yarn
+yarn create toolpad-app --example core-tutorial
+```
+
+  </codeblock>
+
+2. To start the basic project on [http://localhost:3000](http://localhost:3000/), run:
+
+<codeblock storageKey="package-manager">
+
+```bash npm
+cd <project-name>
+npm install && npm run dev
+```
+
+```bash pnpm
+cd <project-name>
+pnpm install && pnpm dev
+```
+
+```bash yarn
+cd <project-name>
+yarn && yarn dev
+```
+
+</codeblock>
+
+3. The following splash screen appears:
+
+{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/core/installation-1.png", "alt": "Toolpad Core entry point", "caption": "Starting with Toolpad Core", "zoom": true, "indent": 1 }}
+
+4. The app has one page already created, which can be viewed by clicking on the "Go to page" button. The following page should appear:
+
+{{"demo": "TutorialDefault.js", "iframe": true, "hideToolbar": true }}
+
+## Create a new page
+
+<br/>
+
+1. To add a new page and make it appear in the sidebar navigation, create a new folder within the `(dashboard)` directory titled `page-2` and add the following content to `page.tsx` inside it:
+
+```tsx
+import { Typography, Container } from '@mui/material';
+export default function Home() {
+  return (
+    <main>
+      <Container sx={{ mx: 4 }}>
+        <Typography variant="h6" color="grey.800">
+          This is page 2!
+        </Typography>
+      </Container>
+    </main>
+  );
+}
+```
+
+2. Add the newly created page to the sidebar navigation by adding the following code to the navigation items array in `app/layout.tsx`:
+
+```tsx
+const NAVIGATION: Navigation = [
+  {
+    kind: 'header',
+    title: 'Main items',
+  },
+  {
+    slug: '/page',
+    title: 'Page',
+    icon: <DashboardIcon />,
+  },
+  // Add the following new item:
+  {
+    slug: '/page-2',
+    title: 'Page 2',
+    icon: <DashboardIcon />,
+  },
+];
+```
+
+The newly created page can now be navigated to from the sidebar, like the following:
+
+{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/core/tutorial-2.gif", "alt": "Toolpad Core new page", "caption": "Adding pages to navigation", "zoom": true, "indent": 1 }}
```

Comment:
Same for this one?

Reply:
There is no easy way to represent the change in the page content alongside the navigation, which is what the GIF depicts

Reply:
Can't the user just click on the different navigation entries? We'll have to create router adapter for this for the demo, not sure whether that's already possible at the moment. But I suspect it could look a bit like

```tsx
// in-memory router for demos

const [pathname, setPathname] = React.useState(initialPathname);
const [searchParams, setSearchParams] = React.useState(new URLSearchParams());
const navigate = React.useCallback((url, options) => {
  const {pathname, searchParams} = new URL(url, window.location.href);
  setPathname(pathname);
  setSearchParams(searchParams);
},  [])

const routerAdapter = { pathname, searchParams, navigate }

return <AppProvider routerAdapter={routerAdapter}>{/* ... */}</AppProvider>
```

cc @apedroferreira

Reply:
> Can't the user just click on the different navigation entries?

They could; I meant that the dashboard content wouldn't actually change on clicking the different navigation entries in the current behaviour. Agree on using an in-memory router adapter to make navigation work for the demos. 

Perhaps we can merge this and ship an improved version with an in-memory router once that is complete?

Reply:
Sure